src/typing/env_js.ml

```diff
@@ -327,7 +401,7 @@ let string_of_env cx ctx =
 let install_refinement cx x xtypes =
   if exists_in_scope x (peek_env ()) = None then
     let t = SMap.find_unsafe x xtypes in
-    init_env cx x (create_env_entry t t None)
+    init_env cx x (create_env_entry t t None) VarScope (* TODO: verify correct hoist behavior *)
```

Comment:
OK, so I think I have two tests that show the refinement scope needs to depend on the scope of the thing being refined.

Here is why just `VarScope` doesn't work:
```
function scope() {
  {
    let x: {p:?string} = {p:"xxx"};

    if (x.p == null) {
      return;
    }
  }
  x.p // should fail, but $REFI x.p is hoisted to here, so there is no warning
}
```

Here is why just `LexicalScope` doesn't work:
```
function scope() {
  var x: {p:?string} = {p:"xxx"};
  {
    if (x.p == null) {
      return;
    }
  }
  var y: string = x.p // should pass, but refinement doesn't live long enough
}
```

@gabelevi, you helped me understand this refinement stuff in the first place (thanks!). Do you see any problem with my reasoning here? It seems like the scope_kind of the refinement needs to depend on the scope_kind of the thing being refined (`x` in this case).

If that's true, we currently don't store that information beyond `init_env`, so I believe I'd need to add scope_kind to the scope_entry struct. (If true, this isn't so bad, because I think we need to add this kind of information to support TDZ use cases anyway.)